### ISSUE TITLE
Errors while calculating `vina_energy_term`

### ISSUE THREAD

**Issue by mchuck** [deepchem/deepchem#2126]
I'm trying to calculate the `vina_energy_term` for a docked ligand, without much success. Does the `vina_energy_term` function work with the value of `M` different than 5? I assume `M` can be of arbitrary size, but the line https://github.com/deepchem/deepchem/blob/bc05e7c116327da590bb1b64a1a63dba7050ee17/deepchem/dock/pose_scoring.py#L200 in `weighted_linear_sum` is raising an exception (w is of shape `(5,)` and x has shape `(5, N, M)`). I believe this line should be changed to:

```python 
return np.tensordot(w, x, axes=1) 
```

The tests in the `test_pose_scoring` only test functions for `M = 5` and they fail for other values of `M`.

Anothier thing is, that the documentation of `weighted_linear_sum` suggests, that its result type is a scalar, but the result is used in `cutoff_filter` which is accepting an `(N, M)` input.

Can anyone help? :)

**Comment by nd-02110114** [comment]
Thank you for reporting this issue!
I also seem the `weighted_linear_sum` implementation is not correct....

And, your modification may be correct and I will fix this issue.

```python

return np.tensordot(w, x, axes=1) 
```

**Comment by rbharath** [comment]
Thanks for the fix @nd-02110114! 

@mchuck Thanks for reporting the issue! The latest nightly build of DeepChem (once it's up for this last change, probably by later today/tomorrow) should now work for you :)
